src/Uno.UI/UI/Xaml/Controls/TextBox/TextBox.Android.cs

```diff
@@ -371,7 +371,18 @@ private void UpdateInputScope(InputScope inputScope)
 
 		partial void OnTextWrappingChangedPartial(DependencyPropertyChangedEventArgs e)
 		{
-			//TODO : see bug #8178
+			if (e.NewValue is TextWrapping textWrapping && !string.IsNullOrEmpty(Text) &&
+				(textWrapping is TextWrapping.Wrap || textWrapping is TextWrapping.WrapWholeWords))
```

Comment:
```suggestion
			if (e.NewValue is TextWrapping textWrapping &&
				(textWrapping is TextWrapping.Wrap || textWrapping is TextWrapping.WrapWholeWords))
```

**Reason for the suggested change (please verify it with a test if possible)**

Assume we have a `TextBox` as follows:

```csharp
var textBox = new TextBox { Text = "", TextWrapping = TextWrapping.NoWrap };
```

After that, we change `TextWrapping` to `TextWrapping.Wrap`, we'll not change anything due to `IsNullOrEmpty` check.

Then, if `Text` was changed to a long text that needs wrapping, we'll end up in a state where we have a long text, and wrapping is requested, but it doesn't happen because `IsSingleLine` is `true`